src/main/java/org/openlmis/buq/service/buq/BottomUpQuantificationService.java

```diff
@@ -218,10 +226,14 @@ public BottomUpQuantification save(BottomUpQuantificationDto bottomUpQuantificat
   private void assignInitialSupervisoryNode(BottomUpQuantification bottomUpQuantification) {
     if (bottomUpQuantification.isApprovable()
             && bottomUpQuantification.getSupervisoryNodeId() == null) {
-      UUID supervisoryNode = supervisoryNodeReferenceDataService.findSupervisoryNode(
+      SupervisoryNodeDto supervisoryNode = supervisoryNodeReferenceDataService.findSupervisoryNode(
               bottomUpQuantification.getProgramId(),
-              bottomUpQuantification.getFacilityId()).getId();
-      bottomUpQuantification.setSupervisoryNodeId(supervisoryNode);
+              bottomUpQuantification.getFacilityId());
+      if (supervisoryNode != null) {
+        bottomUpQuantification.setSupervisoryNodeId(supervisoryNode.getId());
+        return;
+      }
+      bottomUpQuantification.setSupervisoryNodeId(null);
```

Comment:
changes in this method are because of a NullPointer that might've occured if the `supervisoryNode` had been `null`.